mediatek: add support for Acer Predator W6d and Acer Vero W6m

goldwrt opened the pull request:
This patch adds support for two variants of the already supported router
Acer Predator Connect W6: The Acer Predator Connect W6d (W6 without 6 GHz
wifi) and the Acer Connect Vero W6m (W6 without 2.5G eth1 port, usb3 port,
and the 6 on-board gpio RGB LEDs, and with a KTD2026 RGB LED controller
instead of the KTD2061 LED controller of the W6/W6d). Therefore, all
device tree information common to W6, W6d, and W6m is contained in file
`acer-w6-common.dtsi` which is included in the three device specific dts
files.

All suggestions of PR #16410 to improve current W6 support are adopted:
- reading device mac addresses from u-boot environment
- avoiding the detour via the file system to set the mac addresses
- correcting gpio LED labels
- shorter dummy `dm-mod.create` string in bootargs
- enable i2c interface

In addition, the dummy `dm-mod.create` string and the product name “Acer
Predator Connect W6” are corrected, and the LED configuration is changed
to reflect RX/TX activity and boot status.

The device tree for the W6m refers to the KTD202x driver suggested in
PR #16860.

A diff of the current `fdt-1` dtb for the W6 and the `fdt-1` created by
this patch for the W6 proves that they are equivalent except for the
changes described above and some line orders.

This patch has been extensively tested with some W6 and some W6m routers
since snapshot r27993 (up to r28000) without any boot fails or any crashes
occurring.

Signed-off-by: George Oldfort <openwrt@10099.de>

jarledo commented:
Hi @goldwrt, thanks for taking the time and combining these common devices into a common .dtsi file. Have you looked at the compiled .dtb files per device to verify if they're combined correctly? I have two w6d devices I can help use for testing, both have been running great using PR #16410 rebased on top of main for a few weeks.

I see you also enabled I2C interface, have you used it for anything yet? I would be very interested in finding a way to disable the crazy bright "mood LED".

musashino205 commented on target/linux/mediatek/image/filogic.mk:
```diff
@@ -152,11 +152,11 @@ TARGET_DEVICES += acelink_ew-7886cax
   DEVICE_DTS_DIR := ../dts
   DEVICE_DTS_LOADADDR := 0x47000000
-  DEVICE_PACKAGES := kmod-usb3 kmod-mt7915e kmod-mt7916-firmware kmod-mt7986-firmware mt7986-wo-firmware e2fsprogs f2fsck mkf2fs
+  DEVICE_PACKAGES := kmod-usb3 kmod-i2c-gpio kmod-mt7915e kmod-mt7916-firmware kmod-mt7986-firmware mt7986-wo-firmware e2fsprogs f2fsck mkf2fs i2c-tools
```
Why do we need `kmod-i2c-gpio` for `DEVICE_PACKAGES`? The all dts files are not using `i2c-gpio`.

And, `i2c-tools` shouldn't be added if it's not used for the basic usage of the device.

musashino205 commented on target/linux/mediatek/dts/mt7986a-acer-predator-w6d.dts:
```diff
@@ -0,0 +1,146 @@
+		compatible = "gpio-leds";
+
+		led_status_red: led0 {
+			label = "ant0:red";
```
Please convert `label` properties of LEDs to `color` and `function`.

```
color = <LED_COLOR_ID_*>;
function = LED_FUNCTION_*;
function-enumerator = <N>;
```

or 

```
color = <LED_COLOR_ID_*>;
function = "function";
function-enumerator = <N>;
```

And, node names of LEDs should be `led-<N>` format.

https://elixir.bootlin.com/linux/v6.6.59/source/Documentation/devicetree/bindings/leds/common.yaml

BTW, `"<color>:<func>"` format should be used for `label` property.

musashino205 commented:
Please remove the merge commits.

musashino205 commented:
Please remove the merge commits instead of adding revert commits.

jarledo commented:
I rebased this commit on top of `main` (commit 85844cfc5c591d00997ade665340315d54bc5822) and flashed one of my w6d devices today. Everything seems to work well, and thanks for your "sledgehammer" script to disable the mood leds! I don't have a serial device connected now, so I don't have a serial boot log to attach. Anything else I could do to help with testing?

hauke commented:
Could you split this into multiple commits to make the review easier please.

1. commit extraction into target/linux/mediatek/dts/mt7986a-acer-w6-common.dtsi without big changes
2. updates to the device tree
3. add new devices

goldwrt committed:
mediatek: split Acer/W6 dts into W6 common dtsi and device specific dts

In order to prepare OpenWrt support for other Acer W6 devices, this commit
moves all device tree components that are used by all Acer W6/W6e/W6d/W6m
routers from mt7986a-acer-predator-w6.dts to mt7986a-acer-w6-common.dtsi
(new file) and includes this dtsi file in mt7986a-acer-predator-w6.dts.

Minor changes had to be made to the device tree in order to improve clarity
and – notably – to reduce the number of dtc warnings:
- replace (obviously wrong) led@<N> gpio led entities by led-<N>
- remove unnecessary (default-state = "off") gpio led statements
- rename entity “memory” to “memory@0”
- add missing #address-cells and #address-size in /soc/mmc@11230000
- add missing #address-cells and #address-size in /soc/pcie@11280000
- introduce symbols “nvmem” and “swport0” in dtsi (referenced in dts)

The changes were checked with `diff -BEZbdtwy --suppress-common-lines ...`
(comparing two dts files created using old and new fdt-1 blobs again), see
https://github.com/openwrt/openwrt/pull/16861/#issuecomment-2455680020 .

This is the first of four commits into which the original commit was split
to make reviews easier and more targeted.

Signed-off-by: George Oldfort <openwrt@10099.de>
Link: https://github.com/openwrt/openwrt/pull/16861
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

goldwrt committed:
mediatek: make use of Acer Predator Connect W6's u-boot environment

In order to prepare OpenWrt support for other Acer W6 devices and to adapt
the procedure to read and set mac addresses which other devices of the same
target are using (instead of needing an additional script and creating an
additional structure in the file system), this commit
- reads device mac addresses from u-boot environment
- avoids the detour via the file system to set the mac addresses
- drops redundant file /lib/preinit/05_extract_factory_data.sh

The idea and the implementation were thankfully taken from PR #16410.

This is the second of four commits into which the original commit was split
to make reviews easier and more targeted.

Signed-off-by: George Oldfort <openwrt@10099.de>
Link: https://github.com/openwrt/openwrt/pull/16861
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

goldwrt committed:
mediatek: improve device tree for Acer Predator Connect W6

In order to prepare for OpenWrt support other Acer W6 devices and to get
a step further to full hardware support for Acer Predator Connect W6, this
commit
- adjusts the product name ("Acer Predator Connect W6")
- updates gpio LED labels to function/color scheme
- show router status by using first rgb led instead of it's red color only
  (blue: booting/failsafe mode; red: sysupgrade; green: running – was: red)
- changes switch/eth1 led configuration to reflect RX/TX activity and speed
  (green: full 1Gbps/2.5Gbps speed; amber: lower speed; blink: RX/TX)
- shortens dummy dm-mod.create string in bootargs
- enables W6's i2c interface

This is the third of four commits into which the original commit was split
to make reviews easier and more targeted.

Signed-off-by: George Oldfort <openwrt@10099.de>
Link: https://github.com/openwrt/openwrt/pull/16861
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

goldwrt committed:
mediatek: add support for Acer Predator W6d and Acer Vero W6m

This commit adds support for two variants of the already supported router
Acer Predator Connect W6: The Acer Predator Connect W6d (W6 without 6 GHz
wifi) and the Acer Connect Vero W6m (W6 without 2.5G eth1 port, usb3 port,
and the 6 on-board gpio RGB LEDs, and with a KTD2026 RGB LED controller
instead of the KTD2061 LED controller of the W6/W6d).

The device tree for the W6m refers to the KTD202x driver suggested in
PR #16860.

Patching target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
removes the code repetition in (old) lines 121 to 124 on the occasion.

This is the last of four commits into which the original commit was split
to make reviews easier and more targeted.

Signed-off-by: George Oldfort <openwrt@10099.de>
Link: https://github.com/openwrt/openwrt/pull/16861
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

jarledo commented:
Hi @goldwrt, I tested your latest split commits. This time I installed it on both my w6d devices, one with a 2022.X uboot version and the other one with a 2023.X. I didn't spot any errors in the system logs so far, and everything seems to be working well. Thanks for your contribution!
